### PR TITLE
Brb Caching

### PR DESCRIPTION
**Pull request by connorreinhold**
## Overview

This PR caches the user's BRB/ user history data and allows them to immediately view it upon opening the app (upon which it will then be updated with new info after a few seconds), removing the previous need for us to auto-login them each time they open the app. This PR also completely removes auto-login option for the user, having them no remained login within the app (even upon relaunch) until they press the logout button

## Changes Made

1) Built up a caching system using Internal Storage and Enums, should be used for eventually caching Eatery data (quite simple to add on, I will address this shortly after)
2) Used this caching system to cache user data upon logging into Get within the app. Removed the need for Eatery to wait for a response from Get's servers before being able to display brb account information, etc.
3) This change allowed me to simplify a lot of the login logic, including removing the need for checking the box to "auto-login" on start

Sorry that this PR is a little big. There was a bit of moving around functions regarding login that caused large changes to the code. The important files to look at here are InternalStorage and the autoLogin method of GetLoginUtilities
## Test Coverage

Have tested as thoroughly as possible on simulators. Need @ningning621 or anyone with an Android device to test a dece amount if possible - this functionality has a lot of potential edge cases with poor WiFi / mobile / things that I cannot simulate.

Mostly play-testing.

## Next Steps

Add caching of eateries for when poor reception exists. Remove the need to depend on JSON, will simplify MainActivity without that huge processJson class
This reminded me that there needs to be a lot more null pointer exception catching in Android Eatery. This is something that I will look into soon. 

</details>

### CODE REVIEW

**Comment by ningning621** [comment]
Haven't had the time to thoroughly glance thru the code yet, but is there a reason why you chose to use internal storage instead of `SharedPreferences`? 

It seems like internal storage is usually used for large amt of data whereas `SharedPreferences` is used for simple key-value pairs, which I'm assuming is what is being stored rn: [here](https://stackoverflow.com/questions/11222499/share-preferences-or-internal-file-storage-any-memory-issue)

File: app/src/main/java/com/cornellappdev/android/eatery/loginviews/LogoutFragment.java [hunk 1]
```diff
@@ -20,12 +18,15 @@
  */
 import com.cornellappdev.android.eatery.MainActivity;
 import com.cornellappdev.android.eatery.R;
+import com.cornellappdev.android.eatery.model.enums.CacheType;
 import com.cornellappdev.android.eatery.presenter.AccountPresenter;
+import com.cornellappdev.android.eatery.util.InternalStorage;
+
```
Comment (connorreinhold) [inline app/src/main/java/com/cornellappdev/android/eatery/loginviews/LogoutFragment.java]: Linting

**Comment by connorreinhold** [comment]
SharedPreferences is used to store key/value pairs and the value has to be a String. So if we wanted to store the BRBInfo model using SharedPreferences we'd need to use Gson to convert the model to a String and back again. With Internal Storage all that is done automatically, so writing/reading objects (especially with other nested objects) is pretty easy.

**Comment by ningning621** [comment]
> SharedPreferences is used to store key/value pairs and the value has to be a String. So if we wanted to store the BRBInfo model using SharedPreferences...

ah i see, i misunderstood the purpose of this pr. in this case, i agree that internal storage is the right call.

**Comment by ningning621** [comment]
Addresses #142
